core/stateful_precompile_test.go

```diff
@@ -397,158 +394,90 @@ func TestContractNativeMinterRun(t *testing.T) {
-			setupState: func(state *state.StateDB) {
-				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
-			},
 			expectedRes: common.Hash(precompile.AllowListNoRole).Bytes(),
-			assertState: func(t *testing.T, state *state.StateDB) {
-				res := precompile.GetContractNativeMinterStatus(state, adminAddr)
-				assert.Equal(t, precompile.AllowListAdmin, res)
-			},
+			assertState: func(t *testing.T, state *state.StateDB) {},
 		},
-		"set enabled": {
-			caller:         adminAddr,
+		"read from noRole address readOnly enabled": {
+			caller:         noRoleAddr,
 			precompileAddr: precompile.ContractNativeMinterAddress,
 			input: func() []byte {
-				input, err := precompile.PackModifyAllowList(noRoleAddr, precompile.AllowListEnabled)
-				if err != nil {
-					panic(err)
-				}
-				return input
-			},
-			suppliedGas: precompile.ModifyAllowListGasCost,
-			readOnly:    false,
-			setupState: func(state *state.StateDB) {
-				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
-			},
-			expectedRes: []byte{},
-			assertState: func(t *testing.T, state *state.StateDB) {
-				res := precompile.GetContractNativeMinterStatus(state, adminAddr)
-				assert.Equal(t, precompile.AllowListAdmin, res)
-
-				res = precompile.GetContractNativeMinterStatus(state, noRoleAddr)
-				assert.Equal(t, precompile.AllowListEnabled, res)
+				return precompile.PackReadAllowList(noRoleAddr)
 			},
+			suppliedGas: precompile.ReadAllowListGasCost,
+			readOnly:    true,
+			expectedRes: common.Hash(precompile.AllowListNoRole).Bytes(),
+			assertState: func(t *testing.T, state *state.StateDB) {},
 		},
-		"mint": {
+		"read from noRole address with insufficient gas": {
 			caller:         noRoleAddr,
 			precompileAddr: precompile.ContractNativeMinterAddress,
 			input: func() []byte {
-				input, err := precompile.PackMintInput(noRoleAddr, common.Big1)
-				if err != nil {
-					panic(err)
-				}
-				to, amount, err := precompile.UnpackMintInput(input[4:])
-				if err != nil {
-					panic(err)
-				}
-				assert.Equal(t, noRoleAddr, to)
-				assert.Equal(t, common.Big1, amount)
-				return input
+				return precompile.PackReadAllowList(noRoleAddr)
 			},
-			suppliedGas: precompile.MintGasCost,
+			suppliedGas: precompile.ReadAllowListGasCost - 1,
 			readOnly:    false,
-			setupState: func(state *state.StateDB) {
-				precompile.SetContractNativeMinterStatus(state, noRoleAddr, precompile.AllowListEnabled)
-				initialBalance := state.GetBalance(noRoleAddr)
-				assert.Equal(t, common.Big0, initialBalance)
-			},
-			expectedRes: []byte{},
-			assertState: func(t *testing.T, state *state.StateDB) {
-				res := precompile.GetContractNativeMinterStatus(state, noRoleAddr)
-				assert.Equal(t, precompile.AllowListEnabled, res)
-
-				finalBalance := state.GetBalance(noRoleAddr)
-				assert.Equal(t, common.Big1, finalBalance)
-			},
+			expectedErr: vmerrs.ErrOutOfGas.Error(),
 		},
-		"mint big": {
```

Comment:
why have we removed this test-case? I'd prefer to test mint capabilities here instead of allowList capabilities. (or preferably both of them).

Reply:
Added it back in. I didn't think this was really different than the other minting test case.